sormas-ui/src/main/java/de/symeda/sormas/ui/vaccination/list/VaccinationListComponent.java

```diff
@@ -63,67 +65,63 @@ public VaccinationListComponent(
 		VaccinationListCriteria criteria,
 		RegionReferenceDto region,
 		DistrictReferenceDto district,
-		boolean showCreateButton,
-		Runnable refreshCallback) {
+		AbstractDetailView<? extends ReferenceDto> view) {
 
 		this(
 			criteria,
 			region,
 			district,
-			showCreateButton,
-			refreshCallback,
 			maxDisplayedEntries -> FacadeProvider.getVaccinationFacade()
-				.getEntriesListWithRelevance(caseReferenceDto, criteria, 0, maxDisplayedEntries));
+				.getEntriesListWithRelevance(caseReferenceDto, criteria, 0, maxDisplayedEntries),
+			view);
 	}
 
 	public VaccinationListComponent(
 		ContactReferenceDto contactReferenceDto,
 		VaccinationListCriteria criteria,
 		RegionReferenceDto region,
 		DistrictReferenceDto district,
-		boolean showCreateButton,
-		Runnable refreshCallback) {
+		AbstractDetailView<? extends ReferenceDto> view) {
 
 		this(
 			criteria,
 			region,
 			district,
-			showCreateButton,
-			refreshCallback,
 			maxDisplayedEntries -> FacadeProvider.getVaccinationFacade()
-				.getEntriesListWithRelevance(contactReferenceDto, criteria, 0, maxDisplayedEntries));
+				.getEntriesListWithRelevance(contactReferenceDto, criteria, 0, maxDisplayedEntries),
+			view);
 	}
 
 	public VaccinationListComponent(
 		EventParticipantReferenceDto eventParticipantReferenceDto,
 		VaccinationListCriteria criteria,
 		RegionReferenceDto region,
 		DistrictReferenceDto district,
-		boolean showCreateButton,
-		Runnable refreshCallback) {
+		AbstractDetailView<? extends ReferenceDto> view) {
 
 		this(
 			criteria,
 			region,
 			district,
-			showCreateButton,
-			refreshCallback,
 			maxDisplayedEntries -> FacadeProvider.getVaccinationFacade()
-				.getEntriesListWithRelevance(eventParticipantReferenceDto, criteria, 0, maxDisplayedEntries));
+				.getEntriesListWithRelevance(eventParticipantReferenceDto, criteria, 0, maxDisplayedEntries),
+			view);
 	}
 
 	private VaccinationListComponent(
 		VaccinationListCriteria criteria,
 		RegionReferenceDto region,
 		DistrictReferenceDto district,
-		boolean showCreateButton,
-		Runnable refreshCallback,
-		Function<Integer, List<VaccinationListEntryDto>> entriesListSupplier) {
+		Function<Integer, List<VaccinationListEntryDto>> entriesListSupplier,
+		AbstractDetailView<? extends ReferenceDto> view) {
 		super(I18nProperties.getString(Strings.entityVaccinations));
 
+		this.view = view;
+
 		VaccinationList vaccinationList = new VaccinationList(criteria.getDisease(), entriesListSupplier);
```

Comment:
This code is a duplicate of the code in the first constructor (l.55 ff). Better call this constructor first and call `createNewVaccinationButton` after.